crates/cow-amm/src/indexer.rs

```diff
@@ -0,0 +1,162 @@
+use {
+    crate::{
+        cow_amm::CowAmm,
+        cow_amm_constant_product_factory::CowAmmConstantProductFactoryHandler,
+    },
+    contracts::cow_amm_constant_product_factory,
+    ethcontract::{common::DeploymentInformation, dyns::DynWeb3, Address},
+    ethrpc::current_block::RangeInclusive,
+    shared::event_handling::EventStoring,
+    std::{
+        collections::{BTreeMap, HashMap},
+        sync::Arc,
+    },
+    tokio::sync::RwLock,
+};
+
+pub type CowAmmRegistry = HashMap<Address, CowAmm>;
+/// Registry with the format: (block number, (log index, event))
+pub type EventsRegistry = BTreeMap<u64, BTreeMap<usize, cow_amm_constant_product_factory::Event>>;
```

Comment:
With the latest revisions to the standard CoW AMMs no longer have to be stateful. That means these types can probably be collapsed into a single mapping from block where the pool was created to the pool itself:
```rust
BTreeMap<u64, CowAmm>
```
Also with this we don't need an `EventsRegistry`, right?

Reply:
@MartinquaXD what happens if there is a reorg and the event `Deployed` specifies a different `amm` address? could that happen?